posthog/async_migrations/runner.py

```diff
@@ -77,23 +79,32 @@ def start_async_migration(migration_name: str, ignore_posthog_version=False) ->
 
     mark_async_migration_as_running(migration_instance)
 
-    return run_async_migration_next_op(migration_name, migration_instance)
+    return run_async_migration_operations(migration_name, migration_instance)
+
 
+def run_async_migration_operations(migration_name: str, migration_instance: Optional[AsyncMigration] = None) -> bool:
+    while True:
+        run_next, success = run_async_migration_next_op(migration_name, migration_instance)
+        if not run_next:
+            return success
 
-def run_async_migration_next_op(migration_name: str, migration_instance: Optional[AsyncMigration] = None, run_all=True):
+
+def run_async_migration_next_op(migration_name: str, migration_instance: Optional[AsyncMigration] = None):
     """
     Runs the next operation specified by the currently running migration
-    If `run_all=True`, we run through all operations recursively, else we run one and return
+    We run the next operation of the migration which needs attention
+
+    Returns (run_next, success)
```

Comment:
Really solid refactor. Very clear what states we can get into with `Returns (run_next, success)`